### ISSUE TITLE
Unexpected e2e failure

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#7212]
https://buildkite.com/sourcegraph/sourcegraph-e2e/builds/962#5d19a4a8-2343-433c-a418-ea4cc0bd1626 failed with a lot of timeouts but also some actual errors:

```

[2019-12-15T07:26:08Z]   ● e2e test suite › Repository component › file tree › expands directory on row click (no navigation)
--
  | [2019-12-15T07:26:08Z]
  | [2019-12-15T07:26:08Z]     expect(received).toEqual(expected) // deep equality
  | [2019-12-15T07:26:08Z]
  | [2019-12-15T07:26:08Z]     Expected: "http://localhost:7080/github.com/sourcegraph/jsonrpc2@c6c7b9aa99fb76ee5460ccd3912ba35d419d493d"
  | [2019-12-15T07:26:08Z]     Received: "https://sourcegraph.com/github.com/sourcegraph/jsonrpc2@c6c7b9aa99fb76ee5460ccd3912ba35d419d493d"
  | [2019-12-15T07:26:08Z]
  | [2019-12-15T07:26:08Z]       296 \|         const url = isAbsolute ? location : this.sourcegraphBaseUrl + location
  | [2019-12-15T07:26:08Z]       297 \|         await retry(async () => {
  | [2019-12-15T07:26:08Z]     > 298 \|             expect(await this.page.evaluate(() => window.location.href)).toEqual(url)
  | [2019-12-15T07:26:08Z]           \|                                                                          ^
  | [2019-12-15T07:26:08Z]       299 \|         })
  | [2019-12-15T07:26:08Z]       300 \|     }
  | [2019-12-15T07:26:08Z]       301 \|


```

My first guess is that my recent PR https://github.com/sourcegraph/sourcegraph/pull/7205 caused this failure, but I don't think that's true because it passed e2e tests before merging (click the **View details** button next to "sqs merged commit d4d0bfc into master" to confirm this).

The e2e tests pass locally. I've seen a lot of e2e flakiness tonight (see https://github.com/sourcegraph/sourcegraph/issues/5721#issuecomment-565778656), and I've already spent a lot of time working through that, so I don't want to waste any more time looking into this myself before getting help from someone who's more familiar with our e2e tests.

Is this just CI/e2e flakiness, or is this an actual problem with the app?

**Comment by sqs** [comment]
I don't think reverting will fix it, so I haven't reverted any of my commits. (The first commit that caused it to break was one that just skipped 2 LSIF non-e2e integration tests, and most of the commits I merged passed the branch e2e tests, which is why I don't think reverting will fix it.)

**Comment by attfarhan** [comment]
This failure is odd because it's saying it expected a localhost address and it got a sourcegraph.com address. I am pretty sure this means it's something to do with the rest of the timeouts you're seeing in the build failures and not something actually broken in the app. 

```
  | [2019-12-15T07:26:08Z]     Expected: "http://localhost:7080/github.com/sourcegraph/jsonrpc2@c6c7b9aa99fb76ee5460ccd3912ba35d419d493d"
  | [2019-12-15T07:26:08Z]     Received: "https://sourcegraph.com/github.com/sourcegraph/jsonrpc2@c6c7b9aa99fb76ee5460ccd3912ba35d419d493d"

```

**Comment by lguychard** [comment]
Watching [the e2e video artifact](https://buildkite.com/organizations/sourcegraph/pipelines/sourcegraph-e2e/builds/962/jobs/5d19a4a8-2343-433c-a418-ea4cc0bd1626/artifacts/902c3f9e-3dae-4e34-a28a-425a840035ec), the test runs on http://localhost:7080 initially, but navigates to https://sourcegraph.com/gorilla/mux/-/blob/mux.go around 4:50, seemingly when running the [`Theme switcher - Changes the theme`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@94abddcf1dc0ed535dd909b3a3d010883f6004cd/-/blob/web/src/e2e/e2e.test.ts#L402-422) test.

This is strange because:
- The test navigates to `mux.go` by calling `driver.page.goto(sourcegraphBaseUrl + '/github.com/gorilla/mux/-/blob/mux.go')`
- `sourcegraphBaseUrl` comes from `getConfig()`, which gets the url from the `SOURCEGRAPH_BASE_URL` environment variable, and AFAICT `sourcegraphBaseUrl` is never mutated (and why would it be mutated only in _some_ test runs?).

**Comment by lguychard** [comment]
This was introduced by https://github.com/sourcegraph/sourcegraph/pull/7175.

Due to the redirect to `/site-admin/external-services/new`, the `beforeAll()` hook fails to add external services:

```
[2019-12-17T20:08:00Z]     TimeoutError: waiting for selector ".e2e-goto-add-external-service-page" failed: timeout 30000ms exceeded
--
  | [2019-12-17T20:08:00Z]
  | [2019-12-17T20:08:00Z]       246 \|         }
  | [2019-12-17T20:08:00Z]       247 \|
  | [2019-12-17T20:08:00Z]     > 248 \|         await (await this.page.waitForSelector('.e2e-goto-add-external-service-page', { visible: true })).click()
  | [2019-12-17T20:08:00Z]           \|                                ^
  | [2019-12-17T20:08:00Z]       249 \|
  | [2019-12-17T20:08:00Z]       250 \|         await (
  | [2019-12-17T20:08:00Z]       251 \|             await this.page.waitForSelector(`[data-e2e-external-service-card-link="${kind.toUpperCase()}"]`,
```

Unfortunately, jest does not exit early when the redirect fails, and tests keep running. As a result, when trying to access `gorilla/mux` on the test instance, the backend [triggers a redirect](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@7539144/-/blob/cmd/frontend/internal/app/ui/handlers.go#L134-143) to [sourcegraph.com](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@7539144/-/blob/cmd/frontend/backend/repos.go#L83-88).

**Comment by lguychard** [comment]
Fixed in #7268
